addons/misra.py

```diff
@@ -1530,6 +1530,40 @@ def misra_10_1(self, data):
                     if e1_et == 'char' and e2_et == 'char':
                         self.reportError(token, 10, 1)
 
+    def misra_10_2(self, data):
+        def isEssentiallySignedOrUnsigned(op):
+            if op and op.valueType and op.valueType.sign:
+                return True
+            return False 
+
+        def isEssentiallyChar(op):
+            if op.isName:
+                return getEssentialType(op) == 'char'
+            return op.isChar
+
+        for token in data.tokenlist:
+            if not token.isArithmeticalOp or not token.str in ['+', '-']:
```

Comment:
Nit; somehow I like `token.str not in ['+', '-']` better it's a bit more explicit. Spontaneously I would not feel 100% sure if `not token.str in ['+', '-'] ` means `(not token.str) in ['+', '-']` or `not (token.str in ['+', '-'])`. Since your code works.. it must mean the latter.

Reply:
I agree. Fixed in latest commit.